lib/bundle/commands/check.rb

```diff
@@ -5,75 +5,37 @@ module Commands
-          puts "The Brewfile's dependencies are satisfied."
-        end
-      end
-
-      def any_casks_to_install?
-        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
-        requested_casks = @dsl.entries.select { |e| e.type == :cask }.map(&:name)
-        requested_casks.any? do |c|
-          !Bundle::CaskInstaller.cask_installed_and_up_to_date?(c)
-        end
-      end
-
-      def any_formulae_to_install?
-        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
-        requested_formulae = @dsl.entries.select { |e| e.type == :brew }.map(&:name)
-        requested_formulae.any? do |f|
-          !Bundle::BrewInstaller.formula_installed_and_up_to_date?(f)
-        end
-      end
+      ARROW = "→"
+      FAILURE_MESSAGE = "brew bundle can't satisfy your Brewfile's dependencies."
 
-      def any_taps_to_tap?
-        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
-        requested_taps = @dsl.entries.select { |e| e.type == :tap }.map(&:name)
-        return false if requested_taps.empty?
-        current_taps = Bundle::TapDumper.tap_names
-        (requested_taps - current_taps).any?
+      def output_errors?
+        ARGV.include?("--verbose")
       end
 
-      def any_apps_to_install?
-        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
-        requested_app_ids = @dsl.entries.select { |e| e.type == :mac_app_store }.map { |e| e.options[:id] }
-        requested_app_ids.any? do |id|
-          !Bundle::MacAppStoreInstaller.app_id_installed_and_up_to_date?(id)
-        end
+      def exit_on_first_error?
+        !ARGV.include?("--verbose")
       end
 
-      def any_formulae_to_start?
-        @dsl ||= Bundle::Dsl.new(Bundle.brewfile)
-        @dsl.entries.select { |e| e.type == :brew }.any? do |e|
-          formula = Bundle::BrewInstaller.new(e.name, e.options)
-          needs_to_start = formula.start_service? || formula.restart_service?
-          next unless needs_to_start
-          next if Bundle::BrewServices.started?(e.name)
+      def run
+        check_result = Bundle::Checker.check(exit_on_first_error?)
 
-          old_names = Bundle::BrewDumper.formula_oldnames
-          old_name = old_names[e.name]
-          old_name ||= old_names[e.name.split("/").last]
-          next if old_name && Bundle::BrewServices.started?(old_name)
+        if check_result.work_to_be_done
+          puts FAILURE_MESSAGE
 
-          true
+          if output_errors?
+            checks = Bundle::Checker::CHECKS
+            check_result.completed_checks.each { |checked| puts "#{checks[checked]} were checked." }
+            check_result.unchecked_checks.each { |unchecked| puts "#{checks[unchecked]} were not checked." }
+            check_result.errors.each { |package| puts "#{ARROW} #{package}" }
+          end
+          puts "Satisfy missing dependencies with `brew bundle install`."
+          exit 1
+        elsif Bundle::Checker.any_formulae_to_start?
+          puts FAILURE_MESSAGE
+          puts "At least one brew formula must be started."
```

Comment:
`At least one service still needs to be started` or something like that, maybe? As-is it feels a bit like "just start any formula('s service) and this will work"

Reply:
✅